### PR TITLE
[TF FE] Add layer test for Round operation

### PR DESCRIPTION
**Pull request by LucaTamSapienza**
### Details:
 - Added tests for Round op for both real and complex tensors
 - Added support for handling complex tensors for the Round operation

### Tickets:
 - #23237

### CODE REVIEW

**Comment by LucaTamSapienza** [comment]
Hi @rkazants,

I'm not too sure about how `TensorFlow` handles the `Round` operation because when I tried it locally passing a complex tensor as input, I didn't get the desired output. I had to round both the complex and real parts, so I followed the same approach for creating the test, but I'm not too sure about its correctness. Could you provide further clarification on this matter? I see on the official tf page that theoretically `Round` is supported for complex tensors, but I'm unsure if it's my problem or tf's.
Here is an example:
```python
>>> import tensorflow as tf
>>> x = tf.constant([1+2j, 3+4j])
>>> rounded = tf.raw_ops.Round(x=x)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/lucatam/Documents/OpenSourceProject/openvino/opvenv/lib/python3.9/site-packages/tensorflow/python/util/tf_export.py", line 403, in wrapper
    return f(**kwargs)
  File "/Users/lucatam/Documents/OpenSourceProject/openvino/opvenv/lib/python3.9/site-packages/tensorflow/python/ops/gen_math_ops.py", line 8770, in round
    _ops.raise_from_not_ok_status(e, name)
  File "/Users/lucatam/Documents/OpenSourceProject/openvino/opvenv/lib/python3.9/site-packages/tensorflow/python/framework/ops.py", line 5883, in raise_from_not_ok_status
    raise core._status_to_exception(e) from None  # pylint: disable=protected-access
tensorflow.python.framework.errors_impl.NotFoundError: Could not find device for node: {{node Round}} = Round[T=DT_COMPLEX128]
All kernels registered for op Round:
  device='XLA_CPU_JIT'; T in [DT_FLOAT, DT_DOUBLE, DT_INT32, DT_INT16, DT_INT8, DT_COMPLEX64, DT_INT64, DT_BFLOAT16, DT_COMPLEX128, DT_HALF]
  device='CPU'; T in [DT_HALF]
  device='CPU'; T in [DT_FLOAT]
  device='CPU'; T in [DT_DOUBLE]
  device='CPU'; T in [DT_BFLOAT16]
  device='CPU'; T in [DT_INT32]
  device='CPU'; T in [DT_INT64]
 [Op:Round] name: 
```
As you can see, TensorFlow raises an error. Instead, if I try to round separately the real and imaginary parts:
```python
>>> import tensorflow as tf
>>> complex_tensor = tf.constant([1.5+2.5j, 3.5+4.5j])
>>> real_part = tf.raw_ops.Real(input=complex_tensor)
>>> imag_part = tf.raw_ops.Imag(input=complex_tensor)
>>> rounded_real_part = tf.raw_ops.Round(x=real_part)
>>> rounded_imag_part = tf.raw_ops.Round(x=imag_part)
>>> rounded_tensor = tf.raw_ops.Complex(rounded_real_part, rounded_imag_part)
>>> print("Original tensor:", complex_tensor.numpy())
Original tensor: [ 1.5+2.5j  3.5+4.5j]
>>> print("Rounded tensor:", rounded_tensor.numpy())
Rounded tensor: [ 2.+2.j  4.+4.j]
```
Everything is okay.

Sorry for the message, but I would like to clarify this aspect. Thank you very much!

Best Regards,

Luca

**Comment by rkazants** [comment]
Hi @LucaTamSapienza,

Great catch. I have the same error. I think it is an issue in TF documentation that they mention complex type: https://www.tensorflow.org/api_docs/python/tf/raw_ops/Round. You can file an issue in their repository or post PR with the fix.

Regarding our GFI, could you please add layer test for Round for the current logic and close GFI then? Will it work for you?

Best regards,
Roman

**Comment by LucaTamSapienza** [comment]
> Great catch. I have the same error. I think it is an issue in TF documentation that they mention complex type: https://www.tensorflow.org/api_docs/python/tf/raw_ops/Round. You can file an issue in their repository or post PR with the fix.

Thanks for the advice! I reported an issue about it on their GitHub.


> Regarding our GFI, could you please add layer test for Round for the current logic and close GFI then? Will it work for you?

Do you mean leaving only the tests for the Round operation, removing all the part about the complex tensor?

**Comment by rkazants** [comment]
> Do you mean leaving only the tests for the Round operation, removing all the part about the complex tensor?

Yes, since TF does not support complex tensor for Round. Only layer test for standard types are needed.

File: tests/layer_tests/tensorflow_tests/test_tf_Round.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            tf_net = sess.graph_def
+
+        ref_net = None
+
+        return tf_net, ref_net
+
+    test_data_basic = [
+        dict(input_shape=shape, input_type=type) 
+        for shape in [[6], [2, 5, 3], [10, 5, 1, 5]]
+        for type in [tf.float32, tf.int32, tf.int64, tf.float64]
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_Round.py:25]: please use `@pytest.mark.parametrize` for each parameter. Here you have two parameters: `input_shape` and `input_type`

File: tests/layer_tests/tensorflow_tests/test_tf_Round.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            dict(input_shape=[10, 5, 1, 5]),
+        ]
+
+    @pytest.mark.parametrize("params", test_data_complex)
+    @pytest.mark.precommit
+    @pytest.mark.nightly
+    def test_complex_round(self, params, ie_device, precision, ir_version, temp_dir, use_legacy_frontend):
+        self._test(*self.create_complex_round_net(**params),
+                ie_device, precision, ir_version, temp_dir=temp_dir,
+                use_legacy_frontend=use_legacy_frontend)
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_Round.py:84]:
```suggestion
```

File: src/frontends/tensorflow_common/src/op/round.cpp [hunk 3]
```diff
@@ -15,12 +15,26 @@ namespace tensorflow {
...
+        // Store the complex part type for the output that will be a complex type tensor
+        element::Type complex_part_type = complex_type_mark->get_complex_part_type();
+
+        input = complex_type_mark->input_value(0);
+        auto round = make_shared<v5::Round>(input, round_mode);
+        set_node_name(node.get_name(), round);
+        auto complex_round = make_shared<ComplexTypeMark>(round, complex_part_type);
+        return {complex_round->output(0)};
+    }
+
```
[the remark this answers is not in the document]
Comment (LucaTamSapienza) [reply src/frontends/tensorflow_common/src/op/round.cpp]: reverted

**Review by rkazants — APPROVED** [review]

**Comment by rkazants** [comment]
build_jenkins

**Comment by LucaTamSapienza** [comment]
@rkazants Perhaps I should edit the title and details of this PR to avoid misunderstandings?

**Comment by rkazants** [comment]
> @rkazants Perhaps I should edit the title and details of this PR to avoid misunderstandings?

I edited the title. GFI title is still fine.

File: tests/layer_tests/tensorflow_tests/test_tf_Round.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+        return tf_net, ref_net
+
+    @pytest.mark.parametrize("input_shape", [[6], [2, 5, 3], [10, 5, 1, 5]])
+    @pytest.mark.parametrize("input_type", [tf.float32, tf.int32, tf.int64, tf.float64])
+    @pytest.mark.precommit
+    @pytest.mark.nightly
+    def test_round_basic(self, input_shape, input_type, ie_device, precision,
+                        ir_version, temp_dir, use_legacy_frontend):
+        self._test(*self.create_tf_round_net(input_shape, input_type),
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_Round.py:27]:
```suggestion
        if ie_device == 'GPU' and input_type in [tf.int32, tf.int64]:
            pytest.skip("GPU error: Requested activation is not supported for integer type")
        self._test(*self.create_tf_round_net(input_shape, input_type),
```

**Comment by rkazants** [comment]
build_jenkins

**Pull request merged**